webboot/wget_test.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"io/ioutil"
+	"strings"
+	"testing"
+)
+
+func TestWget(t *testing.T) {
+	type test struct {
+		name    string
+		isolink string
+		md5link string
+	}
+	tests := []test{
+		{name: "tinyCoreISO", isolink: "http://tinycorelinux.net/10.x/x86_64/release/CorePure64-10.1.iso", md5link: "http://tinycorelinux.net/10.x/x86/release/Core-10.1.iso.md5.txt"},
+		{name: "archLinuxISO", isolink: "http://mirrors.edge.kernel.org/archlinux/iso/2019.08.01/archlinux-2019.08.01-x86_64.iso", md5link: "http://mirrors.edge.kernel.org/archlinux/iso/2019.08.01/md5sums.txt"},
+		{name: "ubuntuISO", isolink: "http://releases.ubuntu.com/bionic/ubuntu-18.04.2-desktop-amd64.iso", md5link: "http://releases.ubuntu.com/bionic/MD5SUMS"},
+	}
+
+	for _, v := range tests {
+		t.Run(v.name, func(t *testing.T) {
+			r, err := wget(v.isolink)
+			if err != nil {
+				t.Errorf("Failed to receive the ISO: got %v, want nil", err)
```

Comment:
I changed the name of wget() to linkOpen() because it can be a little misleading. Since the true features of an actual wget include recursive download, and conversion of links for offline viewing, our version only opens the link and returns an io.ReadCloser which contains a []byte. This []byte then gets utilized inside the write function to copy the information into an actual file. We have split up the work. So I changed the name to linkOpen().